types/mydecimal.go

```diff
@@ -1090,11 +1090,14 @@ func (d *MyDecimal) FromFloat64(f float64) error {
 func (d *MyDecimal) ToFloat64() (f float64, err error) {
 	digitsInt := int(d.digitsInt)
 	digitsFrac := int(d.digitsFrac)
-	// https://en.wikipedia.org/wiki/Double-precision_floating-point_format
-	// The 53-bit significand precision gives from 15 to 17 significant decimal digits precision (2−53 ≈ 1.11 × 10−16).
+	// https://en.wikipedia.org/wiki/Double-precision_floating-point_format#IEEE_754_double-precision_binary_floating-point_format:_binary64
+	// "The 53-bit significand precision gives from 15 to 17 significant decimal digits precision (2−53 ≈ 1.11 × 10−16).
 	// If a decimal string with at most 15 significant digits is converted to IEEE 754 double-precision representation,
-	// and then converted back to a decimal string with the same number of digits, the final result should match the original string.
-	if digitsInt+digitsFrac > 15 {
+	// and then converted back to a decimal string with the same number of digits, the final result should match the original string."
+	// The new method is about 10.5X faster than the old one according to the benchmark in types/mydecimal_benchmark_test.go.
+	// The initial threshold here is 15, we adjusted it to 12 for compatibility.
+	// We did a full test of 12 significant digits to make sure it's correct.
+	if digitsInt+digitsFrac > 12 {
```

Comment:
What's the compatibility problem?

Reply:
The possibility that the implementation of MyDecimal.ToFloat64 is different from the original has not been ruled out. (Although in theory they are equivalent)
Full tests have been done for cases with less than 12 significant digits. So the threshold 12 is safe and correct.